core/modules/query/QueryContext.cc

```diff
@@ -31,19 +31,122 @@
+        if (!dbTblPair.db.empty() && !dbTblPair.table.empty()) {
+            // Get the columns in the table from the DB schema and put them in the tableColumnMap.
+            auto columns = getTableSchema(dbTblPair.db, dbTblPair.table);
+            if (!columns.empty()) {
+                ColumnSet& colSet = tableColumnMap[dbTblPair];
+                for (auto const& col : columns) {
+                    colSet.insert(col);
+                }
+            }
+        }
+    }
+
+    // Use the tableToColumnMap to create the columnToTablesMap.
+    columnToTablesMap.clear();
+    for (auto const& tblToColPair : tableColumnMap) {
+        auto const& dbTbl = tblToColPair.first;
+        auto const& colSet = tblToColPair.second;
+        for (auto const& col : colSet) {
+            DbTableSet& st = columnToTablesMap[col];
+            st.insert(dbTbl);
+        }
+    }
+}
+
+
+std::string QueryContext::columnToTablesMapToString() const {
+    std::string str;
+    for (auto const& elem : columnToTablesMap) {
+        str += elem.first + "( "; // column name
+        DbTableSet const& dbTblSet = elem.second;
+        for (auto const& dbTbl : dbTblSet) {
+            str += dbTbl.db + "." + dbTbl.table + " ";
+        }
+        str += ") ";
+    }
+    return str;
+}
+
+///
+/// Get the table schema from the mysqlSchemaConfig database. Primarily, this is
+/// used to map column names to particular tables.
+std::vector<std::string> QueryContext::getTableSchema(std::string const& dbName,
+                                                      std::string const& tableName) {
+    // Check if it is in our list of table schemas
+    DbTablePair dTPair(dbName, tableName);
+
+    // Get the table schema from the local database.
+    std::vector<std::string> colNames;
+    mysql::MySqlConnection mysqlConn{mysqlSchemaConfig};
+    if (!mysqlConn.connected()) {
+        LOGS(_log, LOG_LVL_ERROR, "QueryContext::getTableSchema failed to connect to schema database!");
+        return colNames; // Reconnection failed. This is an error.
+    }
+
+    // SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name = 'tbl_name' AND table_schema = 'db_name'
+    std::string sql("SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS ");
+    sql += "WHERE table_name = '" + tableName + "' " +
+           "AND table_schema = '" + dbName + "'";
+    LOGS(_log, LOG_LVL_DEBUG, "tableSchema sql=" << sql);
+    int rc = mysql_real_query(mysqlConn.getMySql(), sql.data(), sql.size());
+
+    if (rc) { return colNames; }
+    MYSQL_RES* result = mysql_use_result(mysqlConn.getMySql());
+
+    auto resSchema = mysql::SchemaFactory::newFromResult(result);
+    std::string resSchemaStr;
+    for(auto i=resSchema.columns.begin(), e=resSchema.columns.end(); i != e; ++i) {
+        resSchemaStr += i->name + ", ";
+    }
+    LOGS(_log, LOG_LVL_DEBUG, "tableSchema resSchema=" << resSchemaStr);
```

Comment:
If above code is only used for debugging can you move it inside `if (LOG_CHECK_LVL(_log, LOG_LVL_DEBUG))`?